crypto/x509/x509_req.c

```diff
@@ -236,14 +236,12 @@ int X509_REQ_add1_attr_by_NID(X509_REQ *req,
                               int nid, int type,
                               const unsigned char *bytes, int len)
 {
-	if (req == NULL || bytes == NULL) {
-            ERR_raise(ERR_LIB_X509, ERR_R_PASSED_NULL_PARAMETER);
-            return 0;
-	}
-    if (X509at_add1_attr_by_NID(&req->req_info.attributes, nid,
-                                type, bytes, len))
-        return 1;
-    return 0;
+    if (req == NULL || bytes == NULL) {
```

Comment:
Now looking at this particular change in broader context it looks very awkward to me to check for the NULL parameters in just this particular function but not to do it in the other similar functions around. However I am not sure it is really worth it to add the checks as we do not do that universally throughout the code - what was the motivation for this particular change?

I'd recommend dropping it from this PR and if you want to, you could submit a different PR adding the NULL checks everywhere across these functions. But perhaps it should be discussed first on openssl-project ML whether we want such checks universally.

Reply:
As far as i recall, my motivation for adding this check was that during the changes I did to `x509.c` I made a mistake that lead to the `req` argument to be NULL and thus to a crash.
I had learned during the CMP contribution that every public function should validate its input, and so it would certainly be an improvement to do it here. 

Yet I see your point that it appears imbalanced to not do this consistently for related functions. So following your recommendation I've dropped that change from this PR,
also because it is very bulky already anyway.

On this occasion I've squashed the remaining fixup commits.